### PR TITLE
Enable more ManagedHandler tests

### PR DESCRIPTION
**Pull request by stephentoub**
cc: @geoffkizer

### CODE REVIEW

File: src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs [hunk 1]
```diff
@@ -767,15 +767,18 @@ await ReadCharAsync(cancellationToken).ConfigureAwait(false) != '.')
...
 
-                if (await ReadCharAsync(cancellationToken).ConfigureAwait(false) != ' ')
+                // Parse (optional) reason phrase
+                _sb.Clear();
+                char c = await ReadCharAsync(cancellationToken).ConfigureAwait(false);
+                if (c == ' ')
+                {
+                    c = await ReadCharAsync(cancellationToken).ConfigureAwait(false);
+                }
+                else if (c != '\r')
```
Comment (geoffkizer) [inline src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs]: The RFC doesn't seem to allow this, but if we allow it today, I suppose there's no harm here.

**Comment by geoffkizer** [comment]
LGTM

**Comment by stephentoub** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

**Comment by stephentoub** [comment]
@dotnet-bot test outerloop netcoreapp Windows_NT Debug
@dotnet-bot test Outerloop Linux x64 Debug Build
@dotnet-bot test Outerloop OSX x64 Debug Build

**Pull request merged**